hq/app/schedule/IamAudit.scala

```diff
@@ -0,0 +1,64 @@
+package schedule
+
+import com.gu.anghammarad.models.{Notification, AwsAccount => Account}
+import logic.DateUtils
+import model._
+import schedule.IamMessages.createMessage
+import schedule.IamNotifier.createNotification
+import utils.attempt.FailedAttempt
+
+object IamAudit {
+  def makeCredentialsNotification(allCreds: Map[AwsAccount, Either[FailedAttempt, CredentialReportDisplay]]):List[Either[FailedAttempt, Notification]] = {
+    allCreds.toList.map { case (awsAccount, eFCreds) =>
+      eFCreds.map { credsReport =>
+        val outdatedKeys = outdatedKeysInfo(findOldAccessKeys(credsReport))
+        val missingMfa = missingMfaInfo(findMissingMfa(credsReport))
+        val message = createMessage(outdatedKeys, missingMfa)
+        createNotification(Account(awsAccount.id), message)
+      }
+    }
+  }
+
+  def findOldAccessKeys(credsReport: CredentialReportDisplay): CredentialReportDisplay = {
+    val filteredMachines = credsReport.machineUsers.filter(user => hasOutdatedKey(List(user.key1, user.key2)))
+    val filteredHumans = credsReport.humanUsers.filter(user => hasOutdatedKey(List(user.key1, user.key2)))
+    credsReport.copy(machineUsers = filteredMachines, humanUsers = filteredHumans)
+  }
+
+  def hasOutdatedKey(keys: List[AccessKey]): Boolean = keys.exists(key => DateUtils.dayDiff(key.lastRotated).getOrElse(1L) > 90L)
```

Comment:
this function could also check for whether the key is active or not (maybe update the name accordingly) - I don't think initially we should be sending alerts for inactive keys

Reply:
Yes I also thought that initially, but Kate has wants us to alert on disabled keys. They said that keys should either be deleted or rotated, because there is always the chance they will be reenabled 👍🏽